### PR TITLE
Agnes movie-site

### PR DESCRIPTION
**Pull request by aagneshansson**
Link to my deployed project:
https://watchamovie.netlify.app/

My reflections on how this week's project turned out:
Really fun to work in React, want more time to finally create the select (drop-down list) with different movie-lists.

Things I'd like to have clarified or explained in more detail:
If/else statements in React to different pages.

### CODE REVIEW

**Review by AnnaHellqvist — APPROVED** [review]
Really nice page you have created this week Agnes! 😄 🎥 You've managed to fulfill the blue and red requirements, great job! The responsiveness of your startpage is really good, I am kind of jealous, will have a second look at my own now 🙈  Your CSS files look awesome, and I encourage you to work as hard on indentations and to remove empty lines in the js-files as well.  I made some comments and suggestions on how to shorten your code you could take a look at, mostly indentations to fix! Be proud of how far you've come on such a short time 🥳 🌟

File: code/src/App.js [hunk 1]
```diff
@@ -1,9 +1,29 @@
-import React from 'react'
+import React from 'react';
+import { BrowserRouter, Switch, Route } from 'react-router-dom';
+import MovieList from './components/MovieList.js';
+import MovieDetails from './components/MovieDetails.js';
+import { Trending } from 'components/Trending.js';
+import { Upcoming } from 'components/Upcoming.js';
```
Comment (EmOlsz) [inline code/src/App.js:6]: No need to add `.js` extension

File: code/src/components/Error.js [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+export const Error = () => {
+    return (
+        <section className="error">
+            <Link to="/" exact="true">
+            <div className="back-arrow">
+                    
+                    <p classname="error-animation">
+                    <i class="arrow left"></i>
+                        Back to movies
+                        </p>
```
Comment (EmOlsz) [inline code/src/components/Error.js:14]: Wrong indentations

File: code/src/components/Movie.js [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+                <i class="arrow left"></i>
+                    Movies
+                    </p>
+                </div>
+                
+            </Link>
+            
+        <img src={`https://image.tmdb.org/t/p/w1280${poster_path}`}
+        alt={title} 
+        />
```
Comment (EmOlsz) [inline code/src/components/Movie.js:29]: Wrong indentations in this file as well

File: code/src/components/MovieDetails.js [hunk 4]
```diff
@@ -0,0 +1,51 @@
+import React, { useEffect, useState } from 'react';
+import { useParams } from 'react-router-dom';
+
+import { API_KEY } from './key';
+import { Movie } from './Movie';
+import { Error } from './Error';
+import { Loader } from './Loader';
+
+export const MovieDetails = () => {
```
[the remark this answers is not in the document]
Comment (EmOlsz) [reply code/src/components/MovieDetails.js:9]: Good catch Anna!

File: code/src/components/MovieDetails.js [hunk 5]
```diff
@@ -0,0 +1,51 @@
...
+            setMovies(json);
+        })
+        .catch((error) => { 
+        })
+
+    }, [MOVIE_DETAILS]);
+
+    useEffect(() => {
+        setLoading(false);
+    }, [movies]);
```
[the remark this answers is not in the document]
Comment (EmOlsz) [reply code/src/components/MovieDetails.js:29]: We can have as many `useEffect()` hooks as we want, there is no rule regarding that. It's ok as it is

File: code/src/components/MovieDetails.js [hunk 6]
```diff
@@ -0,0 +1,51 @@
...
+
+    if (loading === true) {
```
Comment (EmOlsz) [inline code/src/components/MovieDetails.js:31]: We can simplify this : `if (loading) { ... }`

File: code/src/components/MovieDetails.js [hunk 7]
```diff
@@ -0,0 +1,51 @@
...
+        return (
+            <Loader />
+        )
+
+    } else if (movies.backdrop_path === undefined) {
```
Comment (EmOlsz) [inline code/src/components/MovieDetails.js:36]: `} else if (!movies.backdrop_path) {`

**Pull request closed without merging**